### PR TITLE
Add a WIP azure cli

### PR DESCRIPTION
**Pull request by jeremyfowers**
Checking in a work in progress Azure CLI. Not intended to be used by anyone else yet and not documented. Will file issues under the "azure" label to document future work. 

This PR places the cli under test, but the test only runs if an Azure CLI file has changed.

### CODE REVIEW

File: .github/workflows/test_azure.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  clean_up:
+    needs: build_and_test
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install az cli
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y azure-cli
```
Comment (ramkrishna2910) [inline .github/workflows/test_azure.yml:68]: nit: do we want to specify a version for azure cli?
(Cant be done through package manager, will have to use `wget` followed by `dpkg -i`)

File: src/mlagility/cli/cloud/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+azure-mgmt-resource
+azure-mgmt-compute
+azure-mgmt-network
+azure-identity
```
Comment (ramkrishna2910) [inline src/mlagility/cli/cloud/requirements.txt:4]: Specify versions.

File: src/mlagility/cli/cloud/setup_part_1.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Setup
+
+MYUSER=azureuser
+MYHOME=/home/$MYUSER
+
+# Install NFS
+
+# sudo apt-get -y update
+# sudo apt-get -y install nfs-common
+# sudo mkdir -p /mount/mlagilitystorage/mla-share
```
Comment (ramkrishna2910) [inline src/mlagility/cli/cloud/setup_part_1.sh:10]: Remove dead code

File: src/mlagility/cli/cloud/setup_part_1.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+echo 'export MLAGILITY_DEBUG=True' >> $MYHOME/.bashrc
+# echo 'export ML_CACHE=/mount/mlagilitystorage/mla-share/ml-cache' >> $MYHOME/.bashrc
+# echo 'export HF_DATASETS_CACHE="$ML_CACHE""/huggingface"' >> $MYHOME/.bashrc
+# echo 'export TRANSFORMERS_CACHE="$ML_CACHE""/huggingface"' >> $MYHOME/.bashrc
+# echo 'export HF_HOME="$ML_CACHE""/huggingface"' >> $MYHOME/.bashrc
+# echo 'export XDG_CACHE_HOME="$ML_CACHE""/huggingface"' >> $MYHOME/.bashrc
+# echo 'export TORCH_HOME="$ML_CACHE""/torch-hub"' >> $MYHOME/.bashrc
+# echo 'export TORCH_HUB="$ML_CACHE""/torch-hub"' >> $MYHOME/.bashrc
+# echo 'sudo mount -t nfs mlagilitystorage.file.core.windows.net:/mlagilitystorage/mla-share /mount/mlagilitystorage/mla-share -o vers=4,minorversion=1,sec=sys' >> $MYHOME/.bashrc
```
Comment (ramkrishna2910) [inline src/mlagility/cli/cloud/setup_part_1.sh:47]: Remove commented code

File: src/mlagility/cli/cloud/setup_part_1.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+# conda
+
+wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+bash Miniconda3-latest-Linux-x86_64.sh -b -p $MYHOME/miniconda3
+$MYHOME/miniconda3/bin/conda init bash
```
Comment (ramkrishna2910) [inline src/mlagility/cli/cloud/setup_part_1.sh:53]: Can we add error handling for network based commands to prevent unknown failures?

**Review by ramkrishna2910 — COMMENTED** [review]
Added some comments

File: .github/workflows/test_azure.yml (hunk 1 above)
Comment (jeremyfowers) [reply .github/workflows/test_azure.yml:68]: IMO we should leave it alone. The commands I'm using in this Action are very basic and I wouldn't expect them to be deprecated.

File: src/mlagility/cli/cloud/requirements.txt (hunk 2 above)
Comment (jeremyfowers) [reply src/mlagility/cli/cloud/requirements.txt]: Added `>=` requirements in https://github.com/groq/mlagility/pull/298/commits/f8cb91518d5c2a3a9907ec74687d48d55d63022a

File: src/mlagility/cli/cloud/setup_part_1.sh (hunk 3 above)
Comment (jeremyfowers) [reply src/mlagility/cli/cloud/setup_part_1.sh]: Fixed in https://github.com/groq/mlagility/pull/298/commits/6b4a178790065ac7e312c24618bbebd69009059e

File: src/mlagility/cli/cloud/setup_part_1.sh (hunk 4 above)
Comment (jeremyfowers) [reply src/mlagility/cli/cloud/setup_part_1.sh]: Fix in https://github.com/groq/mlagility/pull/298/commits/6b4a178790065ac7e312c24618bbebd69009059e

File: .github/workflows/test_azure.yml (hunk 1 above)
Comment (jeremyfowers) [reply .github/workflows/test_azure.yml:68]: I think the `sudo apt-get update` implies that we'll be getting something fresh (as opposed to something ancient and bad)

**Review by ramkrishna2910 — APPROVED** [review]

**Pull request merged**